### DISCUSSION TITLE
Webview: is there a good rendering/UI library that integrates nicely with VS Code styling?

### DISCUSSION THREAD

**Discussion by DavyLandman** [microsoft/vscode-discussions#70] [Extension Development QnA] [answered]
We've been making a few UI's inside webviews, for now just plain html manipulation with typescript. We've used `browserify` to package up all dependencies into a single js file to load in the webview.

I was wondering if there is a good UI library/framework that integrates nicely with VSCode styling, and maybe doesn't require very arcane package.json changes (like for example Angular or React seem a bit heavy weight)? Any experiences that would help us with a choice in this regard?

**Comment by gjsjohnmurray** [comment, marked as answer]
Are you aware of https://github.com/microsoft/vscode-webview-ui-toolkit?

**Reply by DavyLandman** [reply]
I was not, thank you, damn sometimes it's hard to find these things, especially since most terms overlap with regular users of VSCode. 

This solve the design language part beautifully.

**Reply by gjsjohnmurray** [reply]
Glad I could help. A link to it is in VS Code docs at https://code.visualstudio.com/api/ux-guidelines/webviews

**Reply by DavyLandman** [reply]
Ouch 🤦🏼‍♂️ I should have checked there.

Any tips for the most bare-bones framework to pick? React/Vue/Svelte?

**Reply by gjsjohnmurray** [reply]
:shrug: maybe someone else will respond.

**Reply by Tommycore** [reply]
> Any tips for the most bare-bones framework to pick? React/Vue/Svelte?

Ben Awad has a video in which he sets up Svelte for his extension: https://www.youtube.com/watch?v=a5DX5pQ9p5M
